core/domain/story_jobs_one_off.py

```diff
@@ -150,3 +150,36 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class MissingStoryMigrationOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """This job is used to delete story commit log and snapshot metadata
+    models for which stody models are missing.
+
+    """
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [
+            story_models.StoryCommitLogEntryModel,
+            story_models.StorySnapshotMetadataModel
+        ]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            return
+
+        model_class_name = item.__class__.__name__
+        if model_class_name == 'StoryCommitLogEntryModel':
+            model_id = item.story_id
+        else:
+            model_id, _ = item.id.rsplit('-', 1)
+        story = story_models.StoryModel.get(model_id, strict=False)
+        if story is None:
+            item.delete()
```

Comment:
Instead of doing this call https://github.com/oppia/oppia/blob/a7646d60695f1cac99510809079cda4817218367/core/domain/story_services.py#L640 from here and make sure that `delete_story` is implemented in a way that it is possible to handle the partially deleted stories.

Reply:
@vojtechjelinek 
Can you point me to an example of using this for story children? Here, we assume that `StoryModel` is already deleted, but its associated `StoryCommitLogEntryModel` and `StorySnapshotMetadataModel` objects are not deleted. Does the `delete_story` function delete story children with deleted parent story?

Reply:
I'm not sure if it works this way, but if not the code should be rewritten in a way that allows this. For example `delete_exploration` works that way, it doesn't assume anything about the existence of the parent or child models and just deletes the models that exist.